improver/wind_calculations/wind_direction.py

```diff
@@ -184,11 +184,8 @@ def complex_to_deg(complex_in: ndarray) -> ndarray:
 
         angle = np.angle(complex_in, deg=True)
 
-        # If angle negative value - add 360 degrees.
-        angle = np.where(angle < 0, angle + 360, angle)
-
-        # If angle == 360 - set to zero degrees.
-        angle = np.where(np.isclose(angle, 360, atol=0.001), 0.0, angle)
+        # Convert angles so they are in the range [0, 360)
+        angle = np.mod(angle, 360)
```

Comment:
Yeah - it will be interesting to hear what the Met Office reason for this is. I read it as a bug, albeit one which is minor. With the mod function then 360 is an impossible answer, so I reckon this should be sufficient.

Reply:
The only reason that I can think of for the current IMPROVER implementation is due to a concern about machine precision and whether we could get a value so close to 360 that the conversion from float64 to float32 would make the value equal to 360.  With the current implementation, we'd never get 360, but with the implementation in this PR, a value of 359.999985 would go to 360. 

Perhaps this would avoid the issue:

```
angle = np.mod(np.float32(angle), 360)
```

Reply:
Makes sense, changed as suggested.